examples/pytorch/gat/train_ppi.py

```diff
@@ -0,0 +1,104 @@
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import dgl.nn as dglnn
+from dgl.data.ppi import PPIDataset
+from dgl.dataloading import GraphDataLoader
+from sklearn.metrics import f1_score
+
+class GAT(nn.Module):
+    def __init__(self,in_size, hid_size, out_size, heads):
+        super().__init__()
+        self.gat_layers = nn.ModuleList()
+        # three-layer GAT
+        self.gat_layers.append(dglnn.GATConv(in_size, hid_size, heads[0], activation=F.elu))
+        self.gat_layers.append(dglnn.GATConv(hid_size*heads[0], hid_size, heads[1], residual=True, activation=F.elu))
+        self.gat_layers.append(dglnn.GATConv(hid_size*heads[1], out_size, heads[2], residual=True, activation=None))
+        
+    def forward(self, g, inputs):
+        h = inputs
+        for i, layer in enumerate(self.gat_layers):
+            h = layer(g, h)
+            if i == 2:  # last layer 
+                h = h.mean(1)
+            else:       # other layer(s)
+                h = h.flatten(1)
+        return h
+
+def evaluate(g, features, labels, model):
+    model.eval()
+    with torch.no_grad():
+        for layer in model.gat_layers:
+            layer.g = g
+        output = model(g, features)
+        pred = np.where(output.data.cpu().numpy() >= 0, 1, 0)
+        score = f1_score(labels.data.cpu().numpy(), pred, average='micro')
+        return score
+
+def evaluate_in_batches(dataloader, device, model):
+    total_score = 0
+    for batch_id, subgraph in enumerate(dataloader):
+        subgraph = subgraph.to(device)
+        features = subgraph.ndata['feat']
+        labels = subgraph.ndata['label']
+        score = evaluate(subgraph, features, labels, model)
+        total_score += score
+    return total_score / (batch_id + 1) # return average score
+    
+def train(train_dataloader, valid_dataloader, features, device, model):
+    # define train/val samples, loss function and optimizer
```

Comment:
"define train/val samples" does not apply

Reply:
Thanks.